### DISCUSSION TITLE
Why pathname in middleware's dynamic route changed?

### DISCUSSION THREAD

**Discussion by kirillleogky** [vercel/next.js#43774] [Help]
I started to [migrate middleware](https://nextjs.org/docs/messages/middleware-upgrade-guide#executing-middleware-on-internal-nextjs-requests) to new next version (` "next": "^12.3.0" `) and I notice that pathname in this page changed:

![image](https://user-images.githubusercontent.com/47118725/205952728-2a83a005-0935-4669-a520-b30e2cf30139.png)

</br>

`middleware.ts`

![image](https://user-images.githubusercontent.com/47118725/205953339-969e9f7b-1651-488f-9ab9-b87b0f326a5e.png)
```
pathname /_next/static/chunks/pages/admin/manage-creators/edit-creator/%5BcreatorId%5D.js
```

</br>
</br>

before changes:

![image](https://user-images.githubusercontent.com/47118725/205954169-7c11e3a1-466b-4de0-9e1d-7d15c3f6455c.png)

```
pathname /admin/manage-creators/edit-creator/1fb10af8-19c1-4e37-9afb-4aea4a3a7dc7

```

Why pathname in middleware's dynamic route changed and how to do as it was?

**Comment by ciruz** [comment]
Hey in ``12.2`` changes where made ( https://nextjs.org/docs/upgrading#upgrading-to-122 ) with an own Middleware Upgrade Guide ( https://nextjs.org/docs/messages/middleware-upgrade-guide ) for all changes.
